src/mito2/src/compaction.rs

```diff
@@ -150,8 +144,24 @@ impl CompactionScheduler {
         max_parallelism: usize,
     ) -> Result<()> {
         if let Some(status) = self.region_status.get_mut(&region_id) {
-            // Region is compacting. Add the waiter to pending list.
-            status.merge_waiter(waiter);
+            match compact_options {
+                Options::Regular(_) => {
+                    // Region is compacting. Add the waiter to pending list.
+                    status.merge_waiter(waiter);
+                }
+                options @ Options::StrictWindow(_) => {
+                    // Incoming compaction request is manually triggered.
+                    status.set_pending_request(PendingCompaction {
+                        options,
+                        waiter,
+                        max_parallelism,
+                    });
```

Comment:
I wonder if there will be multiple pending requests.

Reply:
> I wonder if there will be multiple pending requests.

The last manual compaction (SWCS) will override previous manual compactions.

Reply:
We may notify the waiter that we will override its operation. Now we just drop the waiter.

Reply:
> We may notify the waiter that we will override its operation. Now we just drop the waiter.

Sounds good.

Reply:
> We may notify the waiter that we will override its operation. Now we just drop the waiter.

f807687041 PTAL @evenyag